rosbridge_library/test/capabilities/test_subscribe.py

```diff
@@ -56,7 +56,7 @@ def test_update_params(self):
             self.assertEqual(subscription.queue_length, min_queue_length)
             self.assertEqual(subscription.fragment_size, min_frag_size)
             self.assertEqual(subscription.compression, "png")
-        except:
+        except:  # noqa: E722  # Will unregister and raise
             subscription.unregister()
             raise
```

Comment:
Should these use `finally:` instead of `except: ... raise`?

Reply:
I think not...

 % `python3`
```python
>>> try:
...     print("Hello")
... finally:
...     raise
...
Hello
Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
RuntimeError: No active exception to reraise
```

Reply:
sorry, I meant 
```py
try:
  ...
finally:
  subscription.unregister()
```

Reply:
I think we are good the way it is.  If failure unregister and raise else unregister (the line after) does what we want.

Reply:
Oh...
```python
try:
    xyz
except:  # noqa: E722  # Will raise
    raise
finally:
    subscription.unregister()
```

Reply:
Is the `except: raise` part necessary? I'm suggesting that you can avoid the `# noqa` by omitting the except clause and only using finally.

Reply:
```
>>> try:
...     Hello
... finally:
...     print("Goodby")
...
Goodby
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
NameError: name 'Hello' is not defined
```